docs/remote_kernels.md

```diff
@@ -0,0 +1,119 @@
+# Remote kernels
+
```

Comment:
💯  on having docs right out of the gate. :) An introductory paragraph is probably a good idea here.

Reply:
Yeah, this isn't done, good idea adding an introduction and maybe add that this is work in progress (e.g. see @bcolloran's recent proposal around remote language plugins).